src/application-confirm/index.jsx

```diff
@@ -1,28 +1,56 @@
-import React from 'react';
-import { Form, Snippet } from '../';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { CheckboxGroup } from '@ukhomeoffice/react-components';
+import { Snippet } from '../';
+
+class ApplicationConfirm extends Component {
+  componentDidMount() {
+    this.setState({
+      submitDisabled: true,
+      values: []
+    });
+  }
+
+  onChange(event) {
+    const declarations = this.props.schema.declarations.options;
+    const checked = event.target.checked;
+    const value = event.target.value;
+
+    this.setState(state => {
+      const values = state.values;
+
+      if (checked) {
+        values.push(value);
+      } else {
+        values.splice(values.indexOf(value), 1);
+      }
 
-const formatters = {
-  declaration: {
-    mapOptions: option => {
       return {
-        ...option,
-        reveal: (
-          <button type="submit" className="govuk-button"><Snippet>actions.submit</Snippet></button>
-        )
+        values,
+        submitDisabled: declarations.length !== values.length
       };
-    }
+    });
   }
-};
-
-export default () => (
-  <Form
-    formatters={formatters}
-    className="application-confirm"
-    model={{
-      declaration: []
-    }}
-    submit={false}
-  >
-    <input type="hidden" name="action" value="submit-pil-application" />
-  </Form>
-);
+
+  render() {
+    return (
+      <div className="application-confirm">
+        <CheckboxGroup
+          name="declarations"
+          label={<Snippet>declarations.title</Snippet>}
+          options={this.props.schema.declarations.options}
+          error={this.props.error}
+          onChange={this.onChange.bind(this)}
+          value={this.state ? this.state.values : []}
+        />
+        <button type="submit" className="govuk-button" disabled={!this.state || this.state.submitDisabled}>
```

Comment:
This looks like the default state will be with a button disabled - ideally the default with no state should be enabled  and only disabled on the client when the state is created.

Reply:
Done.